src/main/scala/mrpowers/jodie/OperationMetricHelper.scala

```diff
@@ -0,0 +1,179 @@
+      version: Long
+  ): Long = {
+    import spark.implicits._
+    val trimmed  = partitionCondition.trim
+    val splitCondition =
+      if (trimmed.contains(" and "))
+        trimmed.split(" and ").toSeq
+      else Seq(trimmed)
+    val conditions = splitCondition.map(x => {
+      val kv = x.split("=")
+      assert(kv.size == 2)
+      s"${kv.head.trim}=${kv.tail.head.trim.stripPrefix("\'").stripSuffix("\'")}"
+    })
+    val jsonSchema = new StructType()
+      .add("numRecords", LongType)
+      .add("minValues", MapType(StringType, StringType))
+      .add("maxValues", MapType(StringType, StringType))
+      .add("nullCount", MapType(StringType, StringType))
+    spark.read
+      .json(FileNames.deltaFile(deltaLog.logPath, version).toString)
+      .withColumn("stats", org.apache.spark.sql.functions.from_json(col("add.stats"), jsonSchema))
+      .select("add.path", "stats")
+      .map(x => {
+        val path = x.getAs[String]("path")
+        conditions.map(x => path != null && path.contains(x)).reduceOption(_ && _) match {
+          case None => 0L
+          case Some(bool) =>
+            if (bool)
+              x.getAs[String]("stats").asInstanceOf[GenericRowWithSchema].getAs[Long]("numRecords")
+            else 0L
+        }
+      })(Encoders.scalaLong)
+      .reduce(_ + _)
+  }
+
+  def transformMetric(
+      metric: Seq[(Long, OperationMetrics)]
+  ): Seq[(Long, Long, Long, Long, Long)] = metric
+    .filter(x =>
+      x._2.isInstanceOf[MergeMetric] || x._2.isInstanceOf[WriteMetric] || x._2
+        .isInstanceOf[DeleteMetric]
+    )
+    .map(x => {
+      val (deleted, inserted, updated, sourceRows) = x._2 match {
+        case MergeMetric(a, b, c, d, e, f, g, h, i, j, k, l) => (b, e, g, j)
+        case WriteMetric(a, b, c)                            => (0L, b, 0L, b)
+        case DeleteMetric(a, b, c, d, e, f, g, h, i, j)      => (a, 0L, 0L, 0L)
+        case _ =>
+          throw new IllegalArgumentException("Other Metric Types are not allowed in this method")
+      }
+      Tuple5.apply(x._1, deleted, inserted, updated, sourceRows)
+    })
+
+  private def filterHistoryByPartition(x: DeltaHistory, partitionCondition: String): Boolean =
+    x.operation match {
+      case "WRITE" => true
+      case "DELETE" | "MERGE" =>
+        if (
+          x.operationParameters
+            .contains("predicate") && x.operationParameters.get("predicate") != None
+        ) {
+          if (partitionCondition.trim.toLowerCase().contains(" and ")) {
+            val conditions = partitionCondition.split(" and ")
+            conditions.map(y => evaluateCondition(x, y)).reduce(_ && _)
+          } else evaluateCondition(x, partitionCondition)
+        } else {
+          false
+        }
+      case _ => false
+    }
+
+  private def evaluateCondition(x: DeltaHistory, partitionCondition: String) = {
+    x.operationParameters.get("predicate").get.contains(partitionCondition.trim)
+  }
+  private def generateMetric(
```

Comment:
Any specific reason for not supporting update operations in this function such as `deltaTable.update(condition,updateValue)`?

Reply:
no specific reason. had missed it. The conditions should be very similar to a delete. have added them now. thanks @brayanjuls for the catch